pandas/tests/series/test_timeseries.py

```diff
@@ -106,6 +106,13 @@ def test_asfreq_resample_set_correct_freq(self):
         # does .resample() set .freq correctly?
         assert df.resample("D").asfreq().index.freq == "D"
 
+    def test_period_dateoffset_conversion(self):
+        ts = pd.date_range('1/1/2012', periods=4, freq=pd.offsets.BQuarterEnd())
+        per = ts.to_period()
+        ts_per = per.to_timestamp()
+
+        tm.assert_datetime_array_equal(ts, ts_per)
```

Comment:
Could use use `tm.assert_index_equal` and rename the variables so it reads `tm.assert_index_equal(result, expected)`